validator_worker/src/follower.rs

```diff
@@ -1,28 +1,44 @@
 use std::error::Error;
 
-use primitives::adapter::Adapter;
+use primitives::adapter::{Adapter, AdapterErrorKind};
 use primitives::validator::{ApproveState, MessageTypes, NewState, RejectState};
 use primitives::BalancesMap;
 
 use crate::core::follower_rules::{get_health, is_valid_transition};
-use crate::heartbeat::heartbeat;
-use crate::sentry_interface::SentryApi;
+use crate::heartbeat::{heartbeat, HeartbeatStatus};
+use crate::sentry_interface::{PropagationResult, SentryApi};
 use crate::{get_state_root_hash, producer};
 use chrono::Utc;
 
-enum InvalidNewState {
+#[derive(Debug)]
+pub enum InvalidNewState {
     RootHash,
     Signature,
     Transition,
     Health,
 }
 
-enum NewStateResult {
-    Ok,
-    Err(InvalidNewState),
+#[derive(Debug)]
+pub enum ApproveStateResult<AE: AdapterErrorKind> {
+    Sent(Vec<PropagationResult<AE>>),
+    /// Conditions for handling the new state haven't been met
+    NotSent,
+    Err {
+        new_state: InvalidNewState,
```

Comment:
This should be `reason` and not `new_state`

Reply:
done